code/modules/projectiles/guns/specialist.dm

```diff
@@ -1245,3 +1247,48 @@ obj/item/weapon/gun/launcher/grenade/update_icon()
 			to_chat(user, SPAN_NOTICE("You unload \the [unloaded_flare] from \the [src]."))
 			user.put_in_hands(unloaded_flare)
 		update_icon()
+
+/obj/item/weapon/gun/flare/unique_action(mob/user)
+	if(!user || !istype(user.loc, /turf/) || !current_mag || !current_mag.current_rounds)
+		return
+		
+	var/turf/flare_turf = user.loc
+	var/area/flare_area = flare_turf.loc
+	
+	if(flare_area.ceiling > CEILING_GLASS)
+		to_chat(user, SPAN_NOTICE("The roof above you is too dense."))
+		return
+	
+	if(!do_after(user, 1 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+		return
+	
+	if(!current_mag.current_rounds) //they fired! Nice try!
+		return
```

Comment:
to avoid this you should also do an if(user.action_busy) check before the do_after